### PR TITLE
[rewrite] Reimplement Gobbie Bag quests using Interaction Framework

### PR DESCRIPTION
**Pull request by Kietsu3756**
Converted "The Gobbiebag Part *" quest line to Interaction Framework.

1. Added "Bowl of Goblin stew 880" to trade list even though it may be unobtainable w/o GM commands currently
2. To reduce redundant code, used a derived GobbieQuest class/meta table from Quest
3. Used fame values from https://www.bg-wiki.com/ffxi/The_Gobbiebag_Part_I ... https://www.bg-wiki.com/ffxi/The_Gobbiebag_Part_X
4. Tested turn in each item individually and the stew with GM commands
5. Added a GM command to ease testing and to set bag size

### CODE REVIEW

**Comment by zach2good** [comment]
Looks pretty good at a glance, I like the quest baseclass! I'll leave the main review to someone else
(I updated the title to be prettier for the changelogs)

File: scripts/quests/jeuno/The_Gobbiebag_Part_I.lua [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+-----------------------------------
+local lowerJeunoID = require("scripts/zones/Lower_Jeuno/IDs")
+-----------------------------------
+
+local GobbieQuest = {}
+
+setmetatable(GobbieQuest, {__index = Quest} )
+GobbieQuest.__index = GobbieQuest
+
+function GobbieQuest:new(params)
```
Comment (claywar) [inline scripts/quests/jeuno/The_Gobbiebag_Part_I.lua]: I feel like the better place to store the base class would be in a helpers.lua file.  It feels a bit awkward for later quests to be calling in the initial quest script.

File: scripts/quests/jeuno/The_Gobbiebag_Part_I.lua [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+    quest.reward = params.reward
+
+    local bagIncrease = 5
+
+    -- If quest is available or accepted, the corect dialogue ID is the expected pre quest inventory size offset by 1
+    local getPendingDialogueId = function (player)
+        return (player:getContainerSize(xi.inv.INVENTORY) + 1)
+    end
+    -- If quest is completed, the corect dialogue ID is the expected post quest inventory size offset by 1
```
Comment (claywar) [inline scripts/quests/jeuno/The_Gobbiebag_Part_I.lua]: Just a styling thing, could you please add a newline between functions?

**Review by claywar — CHANGES_REQUESTED** [review]
In all, this is beautiful work 👍 .  My main request is for moving common procedures out of the first quest, and into a generic helpers file (which appears to be currently supported for quests as well in luautils).  Other than that, a few styling things, and its good to go.  Very nice again!

File: scripts/quests/jeuno/The_Gobbiebag_Part_I.lua (hunk 1 above)
Comment (Kietsu3756) [reply scripts/quests/jeuno/The_Gobbiebag_Part_I.lua]: Yeah... to be honest I have been playing around with this for a while before I finally got it to work properly. Agree with you on the awkwardness factor, I was hoping you guys would tell me a good place to put it :-) Sounds good, l will move and make sure everything works as expected.

File: scripts/quests/jeuno/The_Gobbiebag_Part_I.lua (hunk 2 above)
Comment (Kietsu3756) [reply scripts/quests/jeuno/The_Gobbiebag_Part_I.lua]: Sure, np, also I think I forgot to add , to the last entry in all of the the param tables.

**Comment by Kietsu3756** [comment]
@zach2good, @claywar: Thank you both for the comments and review.

I retested using using the stew with the gm commands for all quests + expected default text once complete. I only tested Gobbiebag I and Gobbiebag X with individual item requests since I did not modify any of the tradeItems tables (except for rename).

It was fun, I learned some Lua and hope to contribute more in the future.

**Comment by claywar** [comment]
I apologize for the confusion.  When I said helpers, I really meant a file called helpers.lua, and not attaching to utils.  We have support in each sub-directory for quests and missions for a specific file to handle that.  For an example, `missions/amk` has one implemented there.  Using a similar format, I think that would be the best place to put the gobbie bag quest common code.

**Comment by Kietsu3756** [comment]
> I apologize for the confusion. When I said helpers, I really meant a file called helpers.lua, and not attaching to utils. We have support in each sub-directory for quests and missions for a specific file to handle that. For an example, `missions/amk` has one implemented there. Using a similar format, I think that would be the best place to put the gobbie bag quest common code.

No problem, done! Thanks for answering my questions on Discord as well :-)

File: scripts/quests/jeuno/The_Gobbiebag_Part_I.lua [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    reward =
+    {
+        fame    = 30,
+        title   = nil,
+    },
+}
+
+local quest = xi.jeuno.helpers.GobbiebagQuest:new(params)
+
+return quest
```
Comment (claywar) [inline scripts/quests/jeuno/The_Gobbiebag_Part_I.lua]: Could you please add a newline at the end of file?

**Review by claywar — COMMENTED** [review]
One last minor change.  Thanks for being patient and asking questions!

File: scripts/quests/jeuno/The_Gobbiebag_Part_I.lua (hunk 3 above)
Comment (Kietsu3756) [reply scripts/quests/jeuno/The_Gobbiebag_Part_I.lua]: I thought my rewrite of the history was flawless... sneaky newline... :-)

**Review by claywar — APPROVED** [review]

**Pull request merged**